src/puppetlabs/trapperkeeper/services/webserver/jetty9_config.clj

```diff
@@ -305,6 +317,54 @@
     (throw (IllegalArgumentException.
              "Error: More than one default server specified in configuration"))))
 
+(defn selectors-count
+  "The number of selector threads that should be allocated per connector per core."
+  [num-cpus]
+  num-cpus)
+
+(defn acceptors-count
+  "The number of acceptor threads that should be allocated per connector per core."
+  [num-cpus]
+  (max 1 (int (/ num-cpus 2))))
+
+(defn threads-per-connector
+  "The total number of threads needed per attached connector."
+  [num-cpus]
+  (+ (acceptors-count num-cpus)
+     (selectors-count num-cpus)))
+
+(schema/defn ^:always-validate
+  connector-count :- schema/Int
+  "Return the number of connectors found in the config."
+  [config :- WebserverRawConfig]
+  (let [connectors [(contains-http-connector? config)
+                    (contains-https-connector? config)]]
+    (count (filter true? connectors))))
+
+(schema/defn ^:always-validate
+  calculate-required-threads :- schema/Int
+  "Calculate the number threads needed to operate based on the number of cores
+  available and the number of connectors present."
+  [config   :- WebserverRawConfig
+   num-cpus :- schema/Int]
+  (+ 1 (* (connector-count config)
+          (threads-per-connector num-cpus))))
+
+(schema/defn ^:always-validate
+  determine-max-threads :- schema/Int
+  "Determine the size of the Jetty thread pool. If the size is specified in the
+  config then use that value, otherwise determine the minimum number of threads
+  needed to operate. If that number is larger than the default then use it,
+  otherwise use the default."
+  [config   :- WebserverRawConfig
+   num-cpus :- schema/Int]
+  (let [calc-max-threads (max (calculate-required-threads config num-cpus)
+                              default-max-threads)]
+    (or (:max-threads config)
```

Comment:
Don't want to be an early optimizer, but would it be worthwhile to move the calculation to the `else` block to avoid the work if it's just going to use the `(:max-threads config)` anyway?

Not a big deal, up to you